packages/core/src/utils/typenames.ts

```diff
@@ -20,11 +19,10 @@ const collectTypes = (obj: EntityLike | EntityLike[], types: string[] = []) => {
     });
   } else if (typeof obj === 'object' && obj !== null) {
     for (const key in obj) {
-      const val = obj[key];
-      if (key === '__typename' && typeof val === 'string') {
-        types.push(val);
+      if (key === '__typename' && typeof obj[key] === 'string') {
```

Comment:
can typeof obj[key] even be no string when we are dealing with a __typename?

Reply:
Technically not. We just haven’t made this assumption for core

Reply:
let's jsut merge as is then :D